### DISCUSSION TITLE
py5_tools.translators example anyone?

### DISCUSSION THREAD

**Discussion by villares** [py5coding/py5generator#62] [Q&A]
Can someone provide me a minimal example of the translator tools in action?

![image](https://user-images.githubusercontent.com/3694604/151707497-5016258b-28e5-42c3-8e3a-ed0237402a29.png)

**Comment by villares** [comment]
Hmmm.... it seems that `translate_token()` does what I thought `translate_code()` would do!

**Comment by hx2A** [comment]
No, you found a bug! Thank you for spotting this.

You should not call `translate_token()` directly, that is for code that has been split up into tokens and works on one token at a time. Your use of `translate_code()` is correct but there is a [bug in that function](https://github.com/hx2A/py5generator/blob/0c5fc721661761f401135d83f5beff96ffcf24de/py5_resources/py5_module/py5_tools/translators/processingpy2imported.py#L63).

```
def translate_code(code):
    util.translate_code(translate_token, code)
```

It should be

```
def translate_code(code):
    return util.translate_code(translate_token, code)
```

I will fix this for the next release. In the mean time, you can hack it with your own function that uses that `util` module or call `translate_file()` or `translate_dir()`. Both work by giving it src and dest Paths for a single file or a directory of files. If it is a directory it will recursively find the relevant files.

**Comment by hx2A** [comment]
Just committed a fix.
